File: mirea/cources/2026-2027/aii-july/05/Part-04-Sorted-two-pointers-and-checklist/examples/example-04.py
# Пример: слияние двух отсортированных массивов в один.
# Ввод: n, m и два массива a и b, уже отсортированные по неубыванию.
# Вывод: объединённый отсортированный массив.

n, m = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

merged = []

i = 0
j = 0
while i < n and j < m:
    if a[i] <= b[j]:
        # Берём меньший текущий элемент из первого массива.
        merged.append(a[i])
        i += 1
    else:
        # Берём меньший текущий элемент из второго массива.
        merged.append(b[j])
        j += 1

while i < n:
    merged.append(a[i])
    i += 1
while j < m:
    merged.append(b[j])
    j += 1

print(*merged)

File: mirea/cources/2026-2027/aii-july/05/Part-04-Sorted-two-pointers-and-checklist/examples/example-04.cpp
// Пример: слияние двух отсортированных массивов в один.
// Ввод: n, m и два массива a и b, уже отсортированные по неубыванию.
// Вывод: объединённый отсортированный массив.

#include <iostream>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, m;
    cin >> n >> m;
    vector<int> a(n), b(m);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    for (int j = 0; j < m; j++) {
        cin >> b[j];
    }

    vector<int> merged;
    merged.reserve(n + m);

    int i = 0;
    int j = 0;
    while (i < n && j < m) {
        if (a[i] <= b[j]) {
            // Берём меньший текущий элемент из первого массива.
            merged.push_back(a[i]);
            i++;
        } else {
            // Берём меньший текущий элемент из второго массива.
            merged.push_back(b[j]);
            j++;
        }
    }

    while (i < n) {
        merged.push_back(a[i]);
        i++;
    }
    while (j < m) {
        merged.push_back(b[j]);
        j++;
    }

    for (int k = 0; k < (int)merged.size(); k++) {
        if (k > 0) {
            cout << ' ';
        }
        cout << merged[k];
    }
    cout << '\n';

    return 0;
}